osquery/tables/system/windows/processes.cpp

```diff
@@ -190,87 +214,125 @@ void genProcess(const WmiResultItem& result, QueryData& results_data) {
+    }
   }
 
-  /// Get the process UID and GID from its SID
-  HANDLE tok = nullptr;
-  std::vector<char> tokUser(sizeof(TOKEN_USER), 0x0);
-  auto ret = OpenProcessToken(hProcess, TOKEN_READ, &tok);
-  if (ret != 0 && tok != nullptr) {
-    unsigned long tokOwnerBuffLen;
-    ret = GetTokenInformation(tok, TokenUser, nullptr, 0, &tokOwnerBuffLen);
-    if (ret == 0 && GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
-      tokUser.resize(tokOwnerBuffLen);
-      ret = GetTokenInformation(
-          tok, TokenUser, tokUser.data(), tokOwnerBuffLen, &tokOwnerBuffLen);
-    }
+  if (context.isAnyColumnUsed({"uid", "gid", "is_elevated_token"})) {
+    /// Get the process UID and GID from its SID
+    HANDLE tok = nullptr;
+    std::vector<char> tokUser(sizeof(TOKEN_USER), 0x0);
+    auto ret = OpenProcessToken(hProcess, TOKEN_READ, &tok);
+    if (ret != 0 && tok != nullptr) {
+      unsigned long tokOwnerBuffLen;
+      ret = GetTokenInformation(tok, TokenUser, nullptr, 0, &tokOwnerBuffLen);
+      if (ret == 0 && GetLastError() == ERROR_INSUFFICIENT_BUFFER) {
+        tokUser.resize(tokOwnerBuffLen);
+        ret = GetTokenInformation(
+            tok, TokenUser, tokUser.data(), tokOwnerBuffLen, &tokOwnerBuffLen);
+      }
 
-    // Check if the process is using an elevated token
-    auto elevated = FALSE;
-    TOKEN_ELEVATION Elevation;
-    DWORD cbSize = sizeof(TOKEN_ELEVATION);
-    if (GetTokenInformation(
-            tok, TokenElevation, &Elevation, sizeof(Elevation), &cbSize)) {
-      elevated = Elevation.TokenIsElevated;
-    }
+      // Check if the process is using an elevated token
+      auto elevated = FALSE;
+      TOKEN_ELEVATION Elevation;
+      DWORD cbSize = sizeof(TOKEN_ELEVATION);
+      if (GetTokenInformation(
+              tok, TokenElevation, &Elevation, sizeof(Elevation), &cbSize)) {
+        elevated = Elevation.TokenIsElevated;
+      }
 
-    r["is_elevated_token"] = elevated ? INTEGER(1) : INTEGER(0);
-  }
-  if (uid != 0 && ret != 0 && !tokUser.empty()) {
-    auto sid = PTOKEN_OWNER(tokUser.data())->Owner;
-    r["uid"] = INTEGER(getUidFromSid(sid));
-    r["gid"] = INTEGER(getGidFromSid(sid));
-  } else {
-    r["uid"] = INTEGER(uid);
-    r["gid"] = INTEGER(gid);
+      r["is_elevated_token"] = elevated ? INTEGER(1) : INTEGER(0);
+    }
+    if (uid != 0 && ret != 0 && !tokUser.empty()) {
+      auto sid = PTOKEN_OWNER(tokUser.data())->Owner;
+      r["uid"] = INTEGER(getUidFromSid(sid));
+      r["gid"] = INTEGER(getGidFromSid(sid));
+    } else {
+      r["uid"] = INTEGER(uid);
+      r["gid"] = INTEGER(gid);
+    }
+    if (tok != nullptr) {
+      CloseHandle(tok);
+      tok = nullptr;
+    }
   }
 
   if (hProcess != nullptr) {
     CloseHandle(hProcess);
   }
-  if (tok != nullptr) {
-    CloseHandle(tok);
-    tok = nullptr;
-  }
+
   results_data.push_back(r);
 }
 
+// collect perf data into a hashmap by pid to later be refferenced
+std::map<std::int32_t, std::map<std::string, std::int64_t>>
```

Comment:
Similarly here, return a `Status` or `void`, and pass in this data structure as a ref or `const` ref if you can.